### PR TITLE
PRMDR-311: Delete doc reference lambda and document service refactor

### PR DESCRIPTION
**Pull request by abbas-khan10**

### CODE REVIEW

File: lambdas/enums/metadata_field_names.py [hunk 1]
```diff
@@ -2,32 +2,30 @@
...
+            DocumentReferenceMetadataFields.ID.value,
+            DocumentReferenceMetadataFields.CONTENT_TYPE.value,
+            DocumentReferenceMetadataFields.CREATED.value,
+            DocumentReferenceMetadataFields.DELETED.value,
+            DocumentReferenceMetadataFields.FILE_NAME.value,
+            DocumentReferenceMetadataFields.FILE_LOCATION.value,
+            DocumentReferenceMetadataFields.NHS_NUMBER.value,
+            DocumentReferenceMetadataFields.TTL.value,
+            DocumentReferenceMetadataFields.TYPE.value,
+            DocumentReferenceMetadataFields.VIRUS_SCANNER_RESULT.value,
```
Comment (joefong-nhs) [inline lambdas/enums/metadata_field_names.py:28]: If we are not skipping some field, I think we can get all the field value like this:
```
        return [field.value for field in DocumentReferenceMetadataFields]
```

File: lambdas/models/document_reference.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+                return file_key
+            raise InvalidDocumentReferenceException(
+                "Failed to parse object key from file location"
+            )
+        except IndexError:
+            raise InvalidDocumentReferenceException(
+                "Failed to parse object key from file location"
+            )
+
+    def create_unique_filename(self, duplicates):
```
Comment (joefong-nhs) [inline lambdas/models/document_reference.py:70]:
```suggestion
    def create_unique_filename(self, duplicates: int):
```
From the usage seems that this `duplicates` is an integer

**Review by joefong-nhs — APPROVED** [review]
Looks great to me! Really appreciate your hard work in migrating the model to pydantic and all the refactoring 😄  
Gave a few minor suggestions, but I think it's good to go without any change :)

File: lambdas/services/document_manifest_service.py [hunk 3]
```diff
@@ -5,7 +5,7 @@
 import zipfile
 
 from botocore.exceptions import ClientError
-from models.document import Document
+from models.document_reference import DocumentReference
```
[the remark this answers is not in the document]
Comment (abbas-khan10) [reply lambdas/services/document_manifest_service.py:8]: Good spot, forgot to delete this after migrating the functionality

**Pull request merged**